### ISSUE TITLE
Implement enriched CLI logger

### ISSUE THREAD

**Issue by matejchalk** [code-pushup/cli#191]
Follow-up to
- #151 
- #190

PoC:
- #189
- implementation: [tools/pocs/stdout-mock.mjs](https://github.com/flowup/quality-metrics-cli/blob/stdout-poc/tools/pocs/stdout-mock.mjs)

### Libraries

- [@poppinss/cliui](https://github.com/poppinss/cliui#readme) 

---

TODO: enrich description once it's prioritized

**Comment by BioPhoton** [comment]
@matejchalk 
We can close this one in favour of #247

**Comment by matejchalk** [comment]
> @matejchalk 
> We can close this one in favour of #247 ?

Good idea to extract the format flag, but that's just a small part of this issue, so I wouldn't close this one.

**Comment by matejchalk** [comment]
The format flag is a tiny part of this issue (I removed from the description just now to prevent confusion). The main focus of this issue is to improve the CLI logging in general, hence the PoC in #189. We decided to do just the report formatting for now (#190), and return to this one later.

[stdout.webm](https://github.com/flowup/quality-metrics-cli/assets/34691111/aa21d957-9c3f-4194-8713-a73061f0d16c)
